### PR TITLE
Add -aws-poll-interval and deprecate -aws-pull-interval

### PR DESCRIPTION
**Pull request by hanshasselberg**

### CODE REVIEW

File: subcommand/sync-catalog/command.go [hunk 1]
```diff
@@ -47,8 +50,13 @@ func (c *Command) init() {
 	c.flags.StringVar(&c.flagConsulServicePrefix, "consul-service-prefix",
 		"", "A prefix to prepend to all services written to Consul from AWS. "+
 			"If this is not set then services will have no prefix.")
-	c.flags.StringVar(&c.flagAWSPullInterval, "aws-pull-interval",
-		"30s", "The interval between fetching from AWS CloudMap. "+
+	c.flags.StringVar(&c.flagAWSDeprecatedPullInterval, "aws-pull-interval",
+		DefaultPollInterval, "[DEPRECATED] The interval between fetching from AWS CloudMap. "+
```
Comment (banks) [inline subcommand/sync-catalog/command.go:54]: Worth saying something like "[DEPRECATED] - use -aws-poll-interval which has the same behaviour"

Comment (banks) [reply subcommand/sync-catalog/command.go:54]: Ah it's below. Probably fine then.

File: subcommand/sync-catalog/command.go [hunk 2]
```diff
@@ -87,12 +95,18 @@ func (c *Command) Run(args []string) int {
 		return 1
 	}
 
+	pollInterval := c.flagAWSPollInterval
+	if pollInterval == DefaultPollInterval && c.flagAWSDeprecatedPullInterval != DefaultPollInterval {
```
Comment (hanshasselberg) [inline subcommand/sync-catalog/command.go:99]: I realized the second check is not necessary, because it doesn't matter if it is default too..

**Review by banks — APPROVED** [review]
Looks good to me!

File: subcommand/sync-catalog/command.go (hunk 2 above)
Comment (hanshasselberg) [reply subcommand/sync-catalog/command.go:99]: Actually it is not! Because only then we now that we want to display the depreciation warning.

**Pull request merged**